manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java

```diff
@@ -187,20 +187,28 @@ public Processor updateProcessor(String bridgeId, String processorId, String cus
         if (!Objects.equals(existingProcessor.getType(), processorRequest.getType())) {
             throw new BadRequestException("It is not possible to update the Processor's Type.");
         }
-        // See https://issues.redhat.com/browse/MGDOBR-516 for updating Action support
-        if (!Objects.equals(existingAction, processorRequest.getAction())) {
-            throw new BadRequestException("It is not possible to update the Processor's Action.");
+        if (existingProcessor.getType() == ProcessorType.SINK) {
+            if (!Objects.equals(existingAction.getType(), processorRequest.getAction().getType())) {
```

Comment:
Why duplicate the if only for the different error message?

Reply:
What would you propose? 

I cannot use the generic term `Gateway` as that is an internal detail. 

`Action` and `Source` are known terms to the User so surely it's better to keep terminology in error messages consistent with their understanding?